src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs

```diff
@@ -12883,6 +12883,90 @@ void M(ref int i)
         }
 
         [Fact]
+        public void CS1510ERR_RefLvalueExpected_02()
+        {
+            var text =
+@"class C
+{
+    void M()
+    {
+        var a = new System.Action<int>(ref x => x = 1);
+        var b = new System.Action<int, int>(ref (x,y) => x = 1);
+        var c = new System.Action<int>(ref delegate (int x) {x = 1;});
+    }
+}
+";
+            DiagnosticsUtils.VerifyErrorsAndGetCompilationWithMscorlib(text,
```

Comment:
@AlekseyTs: I used the existing `CS1510ERR_RefLvalueExpected` test. CS1511 is very different.